core/modules/image/image.module

```diff
@@ -1349,3 +1359,173 @@ function image_filter_keyword($value, $current_pixels, $new_pixels) {
   }
   return (int) $value;
 }
+
+/**
+ * Preprocess function for theme_image_style().
+ *
+ * If theme_image_style() is called without setting explicit height/width for
+ * an SVG image, attempt to set those dimensions. This helps in particular when
+ * uploading an image for the first time in a content form.
+ */
+function image_preprocess_image_style(&$variables) {
+  $uri = $variables['path'];
+  // Only try to get file information if $uri is not empty.
+  if (!empty($uri)) {
+    $is_svg = image_is_svg($uri);
+
+    if ($is_svg && empty($variables['width']) && empty($variables['height'])) {
```

Comment:
Is it possible to have an svg with only one of the width/height dimension defined? If so, then we should be checking if either is empty - not both. Then we should be setting any missing dimension to its default value.

Reply:
This is actually checking if you're calling `theme_image_style()` with neither width nor height, in which case we use defaults. If either one is available, then the image would be scaled to that dimension.